### ISSUE TITLE
[MJAVADOC-798] Consider passing user settings when using invoker

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#1174]
**[Carsten Pfeiffer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carsten.pfeiffer)** opened **[MJAVADOC-798](https://issues.apache.org/jira/browse/MJAVADOC-798?redirect=false)** and commented

I get this in some of our builds:

```
[INFO] The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.7.0:javadoc' has not been previously called for the module: 'de.example:myproject:jar:1.0.0-SNAPSHOT'. Trying to invoke it...
[ERROR] MavenInvocationException: Error when invoking Maven, consult the invoker log file: /tmp/myproject/myproject-submodule/target/invoker/maven-javadoc-plugin1730810899.txt
[WARNING] Creating fake javadoc directory to prevent repeated invocations: /tmp/myproject/myproject/target/site/apidocs
[ERROR] Error fetching link: /tmp/myproject/myproject/target/site/apidocs. Ignored it.
```

And looking at the invoker log, I see that some parent pom cannot be found, because my custom `settings.xml` that I specified with `mvn -s mysettings.xml` was not considered:

```
Invoking Maven for the goals: [org.apache.maven.plugins:maven-javadoc-plugin:3.7.0:javadoc] with no properties

M2_HOME=/home/gis/.m2/wrapper/dists/apache-maven-3.6-bin/5al0cn3ksttl2pq1n2rbut6jh8/apache-maven-3.6.3
MAVEN_OPTS=null
JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64
JAVA_OPTS=null

[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for de.example:myproject-parent:1.0.0-SNAPSHOT: Failure to find de.example.other:someparent:pom:1.9.0-I26 in https://mynexus/content/groups/releases was cached in the local repository, resolution will not be reattempted until the update interval of releases has elapsed or updates are forced and 'parent.relativePath' points at wrong local POM @ de.example:myproject-parent:1.0.0-SNAPSHOT, /tmp/myproject/pom.xml, line 8, column 10
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
```

---

**Affects:** 3.7.0

**Remote Links:**
- [GitHub Pull Request #290
](https://github.com/apache/maven-javadoc-plugin/pull/290)

**Comment by jira-importer** [comment]
**[Carsten Pfeiffer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carsten.pfeiffer)** commented

PR https://github.com/apache/maven-javadoc-plugin/pull/290

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [7e5dea4b3d9789900428f03df23ff8cb18a917a6](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=7e5dea4b3d9789900428f03df23ff8cb18a917a6).
